altair/utils/transformed_data.py

```diff
@@ -0,0 +1,545 @@
+            "Or with conda using:\n"
+            "    conda install -c conda-forge vegafusion-python-embed vegafusion"
+        ) from err
+
+    if isinstance(chart, Chart):
+        # Add dummy mark if None specified to satisfy Vega-Lite
+        if chart.mark == Undefined:
+            chart = chart.mark_point()
+
+    # Deep copy chart so that we can rename marks without affecting caller
+    chart = chart.copy(deep=True)
+
+    # Rename chart or subcharts with magic names that we can look up in the
+    # resulting Vega specification
+    chart_names = name_chart(chart, 0, exclude=exclude)
+
+    # Compile to Vega and extract inline DataFrames
+    with data_transformers.enable("vegafusion-inline"):
+        vega_spec = chart.to_dict(format="vega")
+        inline_datasets = get_inline_datasets_for_spec(vega_spec)
+
+    # Build mapping from mark names to vega datasets
+    facet_mapping = get_facet_mapping(vega_spec)
+    dataset_mapping = get_datasets_for_chart_names(
+        vega_spec, chart_names, facet_mapping
+    )
+
+    # Build a list of vega dataset names that corresponds to the order
+    # of the chart components
+    dataset_names = []
+    for chart_name in chart_names:
+        if chart_name in dataset_mapping:
+            dataset_names.append(dataset_mapping[chart_name])
+        else:
+            raise ValueError("Failed to locate all datasets")
+
+    # Extract transformed datasets with VegaFusion
+    datasets, warnings = runtime.pre_transform_datasets(
+        vega_spec,
+        dataset_names,
+        get_local_tz(),
+        row_limit=row_limit,
+        inline_datasets=inline_datasets,
+    )
+
+    if isinstance(chart, (Chart, FacetChart)):
+        # Return DataFrame (or None if it was excluded) if input was a simple Chart
+        if not datasets:
+            return None
+        else:
+            return datasets[0]
+    else:
+        # Otherwise return the list of DataFrames
+        return datasets
+
+
+def make_magic_chart_name(i: int) -> str:
+    """Make magic chart name for chart number i
+
+    Parameters
+    ----------
+    i : int
+        Mark number
+
+    Returns
+    -------
+    str
+        Mark name
+    """
+    return MAGIC_CHART_NAME.format(i)
+
+
+def name_chart(
+    chart: Union[
+        Chart, FacetChart, LayerChart, HConcatChart, VConcatChart, ConcatChart
+    ],
+    i: int = 0,
+    exclude: Optional[Iterable[str]] = None,
+) -> List[str]:
+    """Name unnamed charts and subcharts
+
+    Name unnamed charts and subcharts so that we can look them up later in
+    the compiled Vega spec.
+
+    Note: This function mutates the input chart by applying names to
+    unnamed charts.
+
+    Parameters
+    ----------
+    chart : Chart, FacetChart, LayerChart, HConcatChart, VConcatChart, or ConcatChart
+        Altair chart to apply names to
+    i : int (default 0)
+        Starting chart index
+    exclude : iterable of str
+        Set of the names of charts to exclude
```

Comment:
```suggestion
        Names of charts to exclude
```
To prevent confusion that it would need to be a Python `set`